src/poll.rs

```diff
@@ -36,7 +36,7 @@ impl<'fd> PollFd<'fd> {
     //
     // // Do something with `pollfd`, which uses the CLOSED fd.
     // ```
-    pub fn new<Fd: AsFd>(fd: &'fd Fd, events: PollFlags) -> PollFd<'fd> {
+    pub fn new<Fd: AsFd + 'fd>(fd: Fd, events: PollFlags) -> PollFd<'fd> {
```

Comment:
If it takes `fd` by value, then we should add a warning to the users in the doc comment to not let them pass the types that are owned file descriptors (e.g., File)

```
    //
    // Different from other I/O-safe interfaces, here, we have to take `AsFd`
    // by reference to prevent the case where the `fd` is closed but it is
    // still in use. For example:
    //
    // ```rust
    // let (reader, _) = pipe().unwrap();
    //
    // // If `PollFd::new()` takes `AsFd` by value, then `reader` will be consumed,
    // // but the file descriptor of `reader` will still be in use.
    // let pollfd = PollFd::new(reader, flag);
    //
    // // Do something with `pollfd`, which uses the CLOSED fd.
    // ```
```

The old comment should also be removed